bin/client.js

```diff
@@ -1,33 +1,106 @@
 #!/usr/bin/env node
-
 "use strict";
 
-var http    = require('http');
-
-// HTTP client to connect to an HTTP server runnign on a
-// local unix socket.
-
-var user = process.env.USER;
-http.get(
-  {
-    socketPath: '/tmp/' + user + '/npg_ranger.sock',
-    path: '/sample?region=1:77970-77980&accession=ERS1060068&format=sam',
-    headers: {'X-Remote-User': user}
-  },
-  function(response) {
-    // Continuously update stream with data
-    var body = '';
-    response.on('data', function(d) {
-      body += d;
-    });
-    response.on('end', function() {
-      // Data reception is done, do whatever with it!
-      console.log('RESPONSE CODE ' + response.statusCode + ' ' + response.statusMessage);
-      console.log('RAW HEADERS ' + response.rawHeaders);
-      console.log('RAW TRAILERS ' + response.rawTrailers);
-      console.log('TRAILER data-truncated' +
-        (JSON.stringify(response.trailers)));
-      console.log("RECEIVED: " + body);
-    });
+const fs    = require('fs');
+const cline = require('commander');
+
+const LOGGER        = require('../lib/logsetup.js');
+const RangerRequest = require('../lib/client/rangerRequest');
+
+/**
+ * @external fs
+ * @see      {@link https://nodejs.org/dist/latest-v4.x/docs/api/fs.html|fs}
+ */
+
+/**
+ * @external commander
+ * @see      {@link https://www.npmjs.com/package/commander|commander}
+ */
+
+/**
+ * Command line client
+ * @module client
+ *
+ * @requires {@link external:fs|fs}
+ * @requires {@link external:commander|commander}
+ * @requires module:logsetup
+ *
+ * Provides a command line client for data retrieval base on GA4GH data sharing
+ * API. Is implemented with parallel, asynchronous requests.
+ *
+ * $ client.js "http://192.168.0.1:5050/resources/AA0011?referenceName=1&start=167856&end=173507&format=BAM" AA0011.bam
+ *
+ * @author Jaime Tovar
+ * @copyright Genome Research Limited 2016
+ */
+
+cline
+  .version('0.2.2')
+  .description('Command line client for GA4GH data streaming')
+  .arguments('<url> <output>')
+  .option('--debug', 'Show debug output')
+  .parse(process.argv);
+
+cline.on('--help', () => {
+  console.log('  Examples:');
+  console.log('');
+  console.log('    $ client.js "http://192.168.0.1:5050/' +
+              'resources/AA0011?referenceName=1&start=167856&end=173507&format=BAM"' +
+              ' AA0011.bam');
+  console.log('');
+});
+
+if ( !cline.args.length || cline.args.length != 2 ) { cline.help(); }
```

Comment:
Would it be helpful to allow for one arg and send the output to the stdout in this case? If/when we have streaming, this would be the main way to use the tool.